test_parse_arn_with_slash_separator sometimes fails.

This test has been _sporadically_ causing failures in Travis-CI recently:

```
======================================================================
FAIL: test_parse_arn_with_slash_separator (util.tests.test_aws.UtilAwsTest)
Assert successful ARN parsing with a slash separator.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/cloudigrade/cloudigrade/cloudigrade/util/tests/test_aws.py", line 83, in test_parse_arn_with_slash_separator
    self.assertEqual(resource_separator, '/')
AssertionError: None != '/'
```

We need to figure out why this may be flaky and fix it.

I presume this is related to:
https://github.com/cloudigrade/cloudigrade/issues/142
https://github.com/cloudigrade/cloudigrade/pull/165

--- a/cloudigrade/util/aws.py
+++ b/cloudigrade/util/aws.py
@@ -101,7 +101,7 @@ class AwsArn(object):
 
     arn_regex = re.compile(r'^arn:(?P<partition>\w+):(?P<service>\w+):'
                            r'(?P<region>\w+(?:-\w+)+)?:'
-                           r'(?P<account_id>\d{12})?:(?P<resource_type>\w+)'
+                           r'(?P<account_id>\d{12})?:(?P<resource_type>[^:/]+)'
                            r'(?P<resource_separator>[:/])?(?P<resource>.*)')
 
     partition = None

--- a/cloudigrade/util/tests/test_aws.py
+++ b/cloudigrade/util/tests/test_aws.py
@@ -85,6 +85,17 @@ class UtilAwsTest(TestCase):
         resource = arn_object.resource
         self.assertIsNotNone(resource)
 
+    def test_parse_arn_with_custom_resource_type(self):
+        """Assert valid ARN when resource type contains extra characters."""
+        mock_arn = 'arn:aws:fakeserv:test-reg-1:012345678901:test.res type:foo'
+        arn_object = AwsArn(mock_arn)
+
+        resource_type = arn_object.resource_type
+        self.assertIsNotNone(resource_type)
+
+        resource = arn_object.resource
+        self.assertIsNotNone(resource)
+
     def test_error_from_invalid_arn(self):
         """Assert error in account ID parsing from a badly-formed ARN."""
         mock_arn = faker.Faker().text()

